### PR TITLE
Add pluggable CredentialFactory strategy

### PR DESCRIPTION
**Pull request by dima767**
For ease of customizing with the existing annotation-driven configuration arrangement.

### CODE REVIEW

**Comment by mmoayyed** [comment]
Look perfect, but I am not sure if compiles?

**Comment by mmoayyed** [comment]
ah nevermind, failure is unrelated.

File: cas-server-support-rest/src/main/java/org/jasig/cas/support/rest/TicketsResource.java [hunk 1]
```diff
@@ -111,20 +115,17 @@
...
-    protected Credential obtainCredential(final MultiValueMap<String, String> requestBody) {
-        return new UsernamePasswordCredential(requestBody.getFirst("username"), requestBody.getFirst("password"));
-    }
+    public static class DefaultCredentialFactory implements CredentialFactory {
 
-    public CentralAuthenticationService getCentralAuthenticationService() {
-        return this.cas;
+        @Override
+        public Credential fromRequestBody(@NotNull MultiValueMap<String, String> requestBody) {
+            return new UsernamePasswordCredential(requestBody.getFirst("username"), requestBody.getFirst("password"));
```
Comment (mmoayyed) [inline cas-server-support-rest/src/main/java/org/jasig/cas/support/rest/TicketsResource.java]: Dima, one other check, if the request does not specify a username/password, could you throw an exception or something similar?

Comment (dima767) [inline cas-server-support-rest/src/main/java/org/jasig/cas/support/rest/TicketsResource.java]: We sure could and the client will receive HTTP 400. But that's an exercise for Monday :-) Happy Thanksgiving!

Comment (mmoayyed) [inline cas-server-support-rest/src/main/java/org/jasig/cas/support/rest/TicketsResource.java]: Kewl! Thanks. Enjoy.

Comment (mmoayyed) [inline cas-server-support-rest/src/main/java/org/jasig/cas/support/rest/TicketsResource.java]: I'll get this merged in for now. If you could do a separate PR for the change, that would be great.

**Pull request merged**